src/api/svg.ts

```diff
@@ -112,38 +117,169 @@ const runnersToPage = (
+
+      // if there's no intersection it means that the line doesn't intersects the svgRect and isn't visible
+      if (intersection.length === 0) return
+
+      if (!isStartInside) {
+        // replace the line start point by the nearest intersection
+        const nearestPoint = intersection.sort((p1, p2) => distanceCoords(start, p1) - distanceCoords(start, p2))[0]
+        start = new PointXY(nearestPoint)
+      }
+
+      if (!isEndInside) {
+        // replace the line start point by the nearest intersection
+        const nearestPoint = intersection.sort((p1, p2) => distanceCoords(end, p1) - distanceCoords(end, p2))[0]
+        end = new PointXY(nearestPoint)
+      }
+    }
+
     page.drawLine({
-      start: {
-        x: element.svgAttributes.x1!,
-        y: element.svgAttributes.y1!,
-      },
-      end: {
-        x: element.svgAttributes.x2!,
-        y: element.svgAttributes.y2!,
-      },
+      start,
+      end,
       thickness: element.svgAttributes.strokeWidth,
       color: element.svgAttributes.stroke,
       opacity: element.svgAttributes.strokeOpacity,
       lineCap: element.svgAttributes.strokeLineCap,
     });
   },
   async path(element) {
+    // the path origin coordinate
+    const basePoint = new PointXY({x: element.svgAttributes.x || 0, y: element.svgAttributes.y || 0})
+    const normalizePoint = (p: PointXY) => new PointXY({ x: p.x - basePoint.x, y:  p.y - basePoint.y})
+
+    /**
+     * 
+     * @param currentPoint is the global point of the current drawing
+     * @param command the path instruction 
+     * @param params the instrction params
+     * @returns the point where the next instruction starts and the new instruction text
+     */
+    const handlePath = (currentPoint: PointXY, command: string, params: number[]) => {
+      switch(command) {
+        case 'm':
+        case 'M':
+          {
+            const nextPoint = new PointXY({
+                x: basePoint.x + params[0],
+                y: basePoint.y + params[1]
+              })
+            return {
+              point: nextPoint,
+              command: `${command}${params[0]},${params[1]}`
+            }
+          }
+        case 'l':
+        case 'L':
+          {
+            const isLocalInstruction = command === 'l'
+            const nextPoint = new PointXY({
+              x: (isLocalInstruction ? currentPoint.x : basePoint.x) + params[0],
+              y: (isLocalInstruction ? currentPoint.y : basePoint.y) + params[1],
+            })
+            const normalizedNext = normalizePoint(nextPoint)
+
+            let endPoint = new PointXY({ x: nextPoint.x, y: nextPoint.y })
+            let startPoint = new PointXY({ x: currentPoint.x, y: currentPoint.y })
+
+            const isStartInside = isCoordinateInsideTheRect(startPoint, svgRect)
+            const isEndInside = isCoordinateInsideTheRect(endPoint, svgRect)
+
+            if (!(isStartInside && isEndInside)) {
```

Comment:
Feels like you are repeating the code from the line. Can't you create a function that takes a line and the viewbox and returns the inner part of the line?